rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/LoggersWindow.java

```diff
@@ -0,0 +1,207 @@
+ *
+ * Changes or newly added records are made in memory and not persisted
+ *
+ */
+
+public class LoggersWindow {
+
+    private Window window;
+
+    private RMController controller;
+
+    private ListGrid loggersGrid;
+
+    public LoggersWindow(RMController controller) {
+        this.controller = controller;
+        this.build();
+    }
+
+    public void show() {
+        this.window.show();
+    }
+
+    public void destroy() {
+        this.window.destroy();
+    }
+
+    private void build() {
+        Layout layout = new VLayout();
+        layout.setMembersMargin(15);
+        layout.setAutoHeight();
+        layout.setMargin(10);
+
+        Label label = new Label("Edit existing loggers levels and/or add a new logger/level. Please note that changes are made in memory and not persisted.");
+        label.setHeight(30);
+
+        loggersGrid = new ListGrid();
+        loggersGrid.setWidth(500);
+        loggersGrid.setHeight(244);
+        loggersGrid.setAlternateRecordStyles(true);
+        loggersGrid.setShowAllRecords(true);
+
+        ListGridField loggerField = new ListGridField("logger", "Logger", 385);
+        loggerField.setRequired(true);
+
+        ListGridField levelField = new ListGridField("level", "Level");
+        levelField.setValueMap("TRACE", "DEBUG", "INFO", "WARN", "ERROR", "FATAL", "OFF");
+        levelField.setCanEdit(true);
+
+        ListGridField removeField = new ListGridField(" ", 20);
+        removeField.setType(ListGridFieldType.IMAGE);
+        removeField.setCanEdit(false);
+        removeField.setCanSort(false);
+        removeField.setDefaultValue("blank.png");
+
+        loggersGrid.setFields(new ListGridField[] { loggerField, levelField, removeField });
+        loggersGrid.setCanResizeFields(true);
+        loggersGrid.setValidateOnChange(true);
+
+        this.controller.fetchLoggersSettings(loggersGrid);
+
+        removeField.addRecordClickHandler(event -> {
+            if (loggersGrid.getRecord(event.getRecordNum())
+                           .getAttribute(" ")
+                           .equals(Images.instance.remove().getSafeUri().asString())) {
+                loggersGrid.removeData(loggersGrid.getRecord(event.getRecordNum()));
+            }
+        });
+
+        IButton addRecordButton = new IButton("Add");
+        addRecordButton.setTooltip("Add a new logger record");
+        addRecordButton.addClickHandler(new ClickHandler() {
+            public void onClick(ClickEvent event) {
+                ListGridRecord newRecord = new ListGridRecord();
+                loggersGrid.addData(newRecord);
+                newRecord.setAttribute("level", "INFO");
+                newRecord.setAttribute(" ", Images.instance.remove().getSafeUri().asString());
+                newRecord.set_canEdit(true);
+                loggersGrid.startEditing(loggersGrid.getRowNum(newRecord));
+                newRecord.setAttribute("logger", "");
+                newRecord.set_canRemove(true);
```

Comment:
I am not sure if set_canRemove(true) is necessary or has any impact on the ListGridRecord, since the clickHandler of the remove field checks if the record has the remove image before removing the record/data

The same for  newLoggerRecord.set_canRemove(false);

Reply:
True, I fixed it !